### PR TITLE
Data-free mixed precision algorithm for Torch and Torch FX backends

### PR DESCRIPTION
**Pull request by alexsu52**
### Changes

Data-free mixed precision algorithm for Torch and Torch FX backends

### Reason for changes

For more accurate weight compression

### Related tickets

ref: 153918

### Tests

NNCF/job/manual/job/post_training_weight_compression/228

### CODE REVIEW

File: nncf/quantization/algorithms/weight_compression/mixed_precision.py [hunk 1]
```diff
@@ -102,20 +102,13 @@ def apply(
...
 
-    @property
-    def available_backends(self) -> List[BackendType]:
-        return [BackendType.OPENVINO]
-
+    @abstractmethod
     def _set_backend_entity(self, model: TModel) -> None:
-        model_backend = get_backend(model)
-        if model_backend == BackendType.OPENVINO:
-            from nncf.quantization.algorithms.weight_compression.openvino_backend import OVMixedPrecisionAlgoBackend
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/mixed_precision.py:112]: is `OVMixedPrecisionAlgoBackend` still needed?

Comment (alexsu52) [reply nncf/quantization/algorithms/weight_compression/mixed_precision.py:112]: Thanks, typo.

File: nncf/quantization/quantize_model.py [hunk 2]
```diff
@@ -511,7 +511,6 @@ def compress_weights(
             )
 
         options = {
-            "sensitivity_metric": sensitivity_metric,
             "awq": awq,
```
Comment (andreyanufr) [inline nncf/quantization/quantize_model.py:514]: What happens if someone pass awq=False, scale_estimation=False, gptq=False ?

File: nncf/quantization/quantize_model.py [hunk 3]
```diff
@@ -511,20 +511,22 @@ def compress_weights(
...
             "scale_estimation": scale_estimation,
             "gptq": gptq,
             "lora_correction": lora_correction,
         }
-        unsupported_options = [name for name, value in options.items() if value is not None]
+        unsupported_options = [name for name, value in options.items() if value]
         if unsupported_options:
             raise nncf.ParameterNotSupportedError(
-                f"Torch backend does not support {', '.join(unsupported_options)} option(s). Set them to None."
+                f"Torch backend does not support {', '.join(unsupported_options)} option(s). Set them to False."
```
Comment (ljaljushkin) [inline nncf/quantization/quantize_model.py:522]: The behavior is changed but not for all backends, I'd align in all places then.

Comment (ljaljushkin) [reply nncf/quantization/quantize_model.py:522]: From the user standpoint, it's a bit confusing that explicit disabling of not supported algorithms is not the same as having them disabled by default. 

On the other hand, default values for some unsupported options might be even more confusing.
For example, for `all_layers` and `backup_mode`. 

So I'd leave the current behavior on develop - no explicit values for unsupported options.

Comment (andreyanufr) [reply nncf/quantization/quantize_model.py:522]: I agree

Comment (alexsu52) [reply nncf/quantization/quantize_model.py]: The last commit was reverted.

**Review by ljaljushkin — APPROVED** [review]

**Review by andreyanufr — APPROVED** [review]

**Pull request merged**